elasticsearch/shard_migration.py

```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python3
+import requests
+import json
+import logging
+from optparse import OptionParser
+
+# Set up logging
+log_format = "[%(asctime)s] [%(levelname)s] %(message)s"
+logging.basicConfig(level=logging.INFO, format=log_format)
+LOG = logging.getLogger(__name__)
+
+HELP_DESCRIPTION = """
+This is a utility script for migrating shards
+in an ElasticSearch cluster from a source node to one of the target nodes.
+"""
+
+HELP_EXAMPLE = """Example: ./indices_migration.py
+ --source-node "store-01.he-eu-hel1.logs.hq"
+ --source-node-ip "10.14.0.64"
+ --target-nodes "store-05.he-eu-hel1.logs.hq,store-04.he-eu-hel1.logs.hq"
+"""
+
+
+def parse_opts():
+    parser = OptionParser(description=HELP_DESCRIPTION, epilog=HELP_EXAMPLE)
+    parser.add_option(
+        "-H",
+        "--host",
+        dest="es_host",
+        default="http://localhost:9200",
+        help="ElasticSearch host.",
+    )
+    parser.add_option(
+        "-S",
+        "--source-node",
+        dest="source_node",
+        type="string",
+        help="Source node name to move shards from.",
+    )
+    parser.add_option(
+        "-I",
+        "--source-node-ip",
+        dest="source_node_ip",
+        type="string",
+        help="IP address of the source node to exclude.",
+    )
+    parser.add_option(
+        "-T",
+        "--target-nodes",
+        dest="target_nodes",
+        type="string",
+        help="Comma-separated list of target node names to move shards to.",
+    )
```

Comment:
In `argparse` you can just allow repeating of the flag using `append` action:
```suggestion
    parser.add_argument(
        "-T",
        "--target-node",
        action='append',
        help="Target node name to move shards to. Repeatable.",
    )
```
Using `dest` is unnecessary since it can just derive the name from the long flag form.